sympy/solvers/ode/single.py

```diff
@@ -852,5 +852,74 @@ def _get_general_solution(self, *, simplify: bool = True):
         return [Eq(lhs, C2 + x), Eq(lhs, C2 - x)]
 
 
+class Lienard(SinglePatternODESolver):
+    r"""
+    The general form of the Lienard ODE:
+
+    .. math :: y''(x) + f(x)*y'(x) + y(x) = 0
+
+    We solve using the fact that all linear second order homogeneous ODEs
+    can be transformed into first order ODEs of Riccati type.
+    Here we solve using the transformation as:
```

Comment:
If this technique works for all linear second order homogeneous ODEs then maybe it should be attempted for more cases than just Lienard ODEs.

Reply:
Let's assume our` linear second-order homogeneous ODE` be `y''(x) + p(x)*y'(x) + q(x)*y(x) = 0`. 

Now for substitution, `b(x) = y'(x)/y(x)`, we can say, ` y''(x) = b'(x)*y(x) + y(x)*b(x)**2`. 

Now substituting back for `y(x), y'(x) and y''(x)`, we get the ode as: `b'(x) + b(x)**2 + p(x)*b(c) + q(x) = 0`.

This is clearly a `Riccati ODE`, but we can't say its solutions always exist, thus we can get the solution only if we can solve this equation. 
If it matches the [special Riccati ODE](https://www.maplesoft.com/support/help/Maple/view.aspx?path=odeadvisor/Riccati). it can have a solution. Moreover not necessarily it should lie in the special Riccati, but we can also solve this ODE, depending on p(x) and q(x) by other methods in `sympy` like for 
`eq = f(x).diff(x)+f(x)**2+2/x*f(x)+1`.

@oscarbenjamin, Kindly share your opinion. 
Thanks.